src/focusAtom.ts

```diff
@@ -49,6 +54,13 @@ export function focusAtom<S, A, R extends void | Promise<void>>(
   ) => O.Lens<S, any, A> | O.Equivalence<S, any, A> | O.Iso<S, any, A>
 ): WritableAtom<A, SetStateAction<A>, R>
 
+export function focusAtom<S, A, R extends void | Promise<void>>(
+  baseAtom: WritableAtom<S | undefined, NonFunction<S>, R>,
+  callback: (
+    optic: O.OpticFor<S | undefined>
+  ) => O.Lens<S, any, A> | O.Equivalence<S, any, A> | O.Iso<S, any, A>
+): WritableAtom<A | undefined, SetStateAction<A>, R>
```

Comment:
curious if you are sure that the return atom value is `A | undefined` instead of `A`.

Reply:
Good catch! Fixed it in [this commit](https://github.com/jotai-labs/jotai-optics/pull/5/commits/6976437315d2040d5775f63a0fe06fd894308f17). I think the other way around would be disallowed in `optics-ts` so you're right